bigquery/api/Snippets/DeleteDataset.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright(c) 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+//
+// [START bigquery_delete_dataset]
+
+using Google.Cloud.BigQuery.V2;
+using System;
+
+public class BigQueryDeleteDataset
+{
+    public void DeleteDataset(
+        string projectId = "your-project-id",
+        string datasetId1 = "your_dataset_without_tables",
+        string datasetId2 = "your_dataset_with_tables"
+    )
+    {
+        BigQueryClient client = BigQueryClient.Create(projectId);
+        // Delete a dataset that does not contain any tables
+        client.DeleteDataset(datasetId: datasetId1);
+        Console.WriteLine($"Dataset {datasetId1} deleted.");
+        // Use the DeleteDatasetOptions to delete a dataset and its contents
+        client.DeleteDataset(
+            datasetId: datasetId2,
+            options: new DeleteDatasetOptions() { DeleteContents = true }
```

Comment:
Would it make sense to split this into two different samples? The [docs](https://cloud.google.com/bigquery/docs/managing-tables#deleting_tables) only have one section for examples, so maybe not.

Reply:
I think this would make sense to split since that would make it more consistent with the other samples (only showing one thing). If you'd like to split it, I'll make separate samples for the languages that have this example and then do a different sample tab section for the second 
 force delete sample.